### PR TITLE
HIVE-21907: Add a new LlapDaemon Management API method to set the daemon capacity

### PR DESCRIPTION
**Pull request by pvary**
3 main changes:
- Test for TestEvictingPriorityBlockingQueue.java so we can be sure it handles queue size change
- Test for TaskExecutorService.java so we can be sure that it will handle capacity changes correctly
- API change for the new method and test for it

### CODE REVIEW

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
 import org.apache.hadoop.hive.llap.daemon.QueryFailedHandler;
 import org.apache.hadoop.hive.llap.daemon.SchedulerFragmentCompletingListener;
 import org.apache.hadoop.hive.llap.daemon.impl.LlapTokenChecker.LlapTokenInfo;
+import org.apache.hadoop.hive.llap.daemon.rpc.LlapDaemonProtocolProtos;
```
Comment (odraese) [inline llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java]: This seems to be an unused import.

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java [hunk 2]
```diff
@@ -176,6 +181,31 @@ public TaskExecutorService(int numExecutors, int waitQueueSize,
     Futures.addCallback(future, new WaitQueueWorkerCallback());
   }
 
+  /**
+   * Sets the TaskExecutorService capacity to the new values. Both the number of executors and the
+   * queue size should be smaller than that original values, so we do not mess up with the other
```
Comment (odraese) [inline llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java]: What other settings? Maybe the comment could be more specific?

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java [hunk 3]
```diff
@@ -176,6 +181,31 @@ public TaskExecutorService(int numExecutors, int waitQueueSize,
...
+   * settings. Setting smaller capacity will not cancel or reject already executing or queued tasks
+   * in itself.
+   * @param newNumExecutors The new number of executors
+   * @param newWaitQueueSize The new number of wait queue size
+   */
+  @Override
+  public synchronized void setCapacity(int newNumExecutors, int newWaitQueueSize) {
+    if (newNumExecutors > configuredMaxExecutors) {
```
Comment (odraese) [inline llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java]: Should we also verify that the new values are >= zero?

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java (hunk 1 above)
Comment (pvary) [reply llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java]: Removed

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java (hunk 2 above)
Comment (pvary) [reply llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java]: Container memory sizes are calculated based on the executors. Adding more executors and this way executing more containers can result in memory oversubscription.

Added more specific comment

File: llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java (hunk 3 above)
Comment (pvary) [reply llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java]: Added check and test

**Pull request closed without merging**